### PR TITLE
Profile page Front End

### PR DESCRIPTION
**Pull request by Abhishek-Geekywolf**

### CODE REVIEW

File: src/ui/src/app/profile/iprofile-from.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    lastName:FormControl<string|null>;
+    gender:FormControl<string|null>;
+    dateofBirth:FormControl<Date|null>;
+    joiningDate:FormControl<Date|null>;
+    phoneNumber:FormControl<string|null>;
+    email:FormControl<string|null>;
+    address:FormControl<string|null>;
+    country:FormControl<string|null>;
+    state:FormControl<string|null>;
+
```
Comment (eldhopjoy) [inline src/ui/src/app/profile/iprofile-from.ts:14]: space and lines

File: src/ui/src/app/profile/profile.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+})
+export class ProfileComponent {
+userForm:FormGroup<IProfileForm>;
+inDate=new Date();
+
+  constructor(private fb:FormBuilder) {
+    this.userForm=this.fb.group({
+      firstName:new FormControl('',Validators.required),
+      lastName:new FormControl(),
+      gender:new FormControl('',Validators.required),
```
Comment (eldhopjoy) [inline src/ui/src/app/profile/profile.component.ts:21]: move this to a fn like buildForm

File: src/ui/src/app/profile/profile.component.html [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+                            Address is required.
+                        </div>
+                        }
+                    </div>
+
+                    <div class="mb-3">
+                        <label for="country" class="form-label">Country</label>
+                        <input type="text-box" class="form-control" id="country" formControlName="country">
+                        @if(userForm.get('country')?.invalid && (userForm.get('country')?.touched ||
+                        userForm.get('country')?.dirty)) {
```
Comment (eldhopjoy) [inline src/ui/src/app/profile/profile.component.html:252]: why you checked touched or dirty logic will the invalid logic is enough to show the message

File: src/ui/src/app/app.routes.ts [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
+        path:'leave-request-history',
+        component:LeaveHistoryComponent
+    },
+    {
+        path:'employee-hierarchy',
+        component:EmployeeHierarchyComponent
+    },
+    {
+        path:'test',
+        component:TestComponent
```
Comment (anilkumar-geeky) [inline src/ui/src/app/app.routes.ts:27]: ???

File: src/ui/src/app/employee-hierarchy/employee-hierarchy.component.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import { Component, OnInit } from '@angular/core';
+import { Employee } from './iemployee-hierarchy';
+import { EmployeeSubordinatesComponent } from './employee-subordinates/employee-subordinates.component';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  selector: 'app-employee-hierarchy',
+  standalone: true,
+  imports: [CommonModule,EmployeeSubordinatesComponent],
```
Comment (anilkumar-geeky) [inline src/ui/src/app/employee-hierarchy/employee-hierarchy.component.ts:9]: remove common module

File: src/ui/src/app/employee-hierarchy/employee-subordinates/employee-subordinates.component.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Component, Input } from '@angular/core';
+import { Employee } from '../iemployee-hierarchy';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  selector: 'app-employee-subordinates',
+  standalone: true,
+  imports: [CommonModule ],
```
Comment (anilkumar-geeky) [inline src/ui/src/app/employee-hierarchy/employee-subordinates/employee-subordinates.component.ts:8]: remove

**Pull request closed without merging**